client/verta/verta/endpoint/nvidia_gpu.py

```diff
@@ -0,0 +1,70 @@
+from enum import Enum
+
+
+class NvidiaGPU:
+    """
+    Nvidia GPU resources
+
+    The JSON equivalent for this is:
+    .. code-block:: json
+
+        {
+            "resources": {"nvidia_gpu": {"number": 1, "model": "V100"}}
+        }
+
+    Parameters
+    ----------
+    number: int > 0
+        Number of GPUs requested to run this endpoint's model.
+    model: NvidiaGPUModel, optional
+        Model of GPU requested to run this endpoint's model. Currently available models are
+        specified in the `NvidiaGPUModel` enum.
+
+    Examples
+    --------
+    .. code-block:: python
+
+        from verta.endpoint.resources import Resources, NvidiaGPU, NvidiaGPUModel
+        resources = Resources(nvidia_gpu=NvidiaGPU(1, NvidiaGPUModel.V100)
+    """
+
+    NUM_GPUS_ERR_MSG = "`number` must be a number greater than 0"
+    MODEL_ERR_MSG = "`model` must be an instance of `verta.endpoint.NvidiaGPUModel`"
+
+    def __init__(self, number, model=None):
+        self._validate_number_of_gpus(number)
+        if model is not None:
+            # convert to NvidiaGPUModel instance if passed in as str
+            model = self._validate_model(model)
+        self.number = number
+        self.model = model
```

Comment:
I'd discourage having `_validate_model()` return because we've had a general pattern of `validate()` methods not returning anything 😬 

I think we could just do

```suggestion
            self._validate_model(model)
        self.number = number
        self.model = NvidiaGPUModel(model)
```

which works if `model` is either an `int` or already an `NvidiaGPUModel`

I think.